### PR TITLE
Tshock

### PR DESCRIPTION
**Pull request by velimir-97**
Added a default config file for tshock(had to be uploaded manually, tshock server files don't include one), by default everything is commented so it doesn't affect anything.
Added ENV variable for config file so location/name can be changed.

### CODE REVIEW

File: tshock/bootstrap.sh [hunk 1]
```diff
@@ -17,20 +17,25 @@ if [ $STORAGETYPE = "mysql" ]; then
 fi
 
 WORLD_PATH="/root/.local/share/Terraria/Worlds/$WORLD_FILENAME"
+if [ ! -f "/config/serverconfig.txt" ]; then
```
Comment (ryansheehan) [inline tshock/bootstrap.sh:20]: Thinking about my comment about the `/config` volume, I think we should place the `serverconfig.txt` in the `/root/.local/share/Terraria/Worlds/` volume.

File: tshock/bootstrap.sh [hunk 2]
```diff
@@ -17,20 +17,25 @@ if [ $STORAGETYPE = "mysql" ]; then
...
+    echo "Server configuration not found, running with default server configuration."
+    echo "Please ensure your desired serverconfig.txt file is volumed into docker: -v <path_to_config_file>:/config"
```
Comment (ryansheehan) [inline tshock/bootstrap.sh:22]: This comment would change if we don't go with the `/config` volume

File: tshock/bootstrap.sh [hunk 3]
```diff
@@ -17,20 +17,25 @@ if [ $STORAGETYPE = "mysql" ]; then
...
 if [ -z "$WORLD_FILENAME" ]; then
   echo "No world file specified in environment WORLD_FILENAME."
-  if [ -z "$@" ]; then 
+  if [ -z "$@" ]; then
     echo "Running server setup..."
   else
     echo "Running server with command flags: $@"
   fi
-  mono --server --gc=sgen -O=all TerrariaServer.exe -configpath "$CONFIGPATH" -logpath "$LOGPATH" "$@" 
+  mono --server --gc=sgen -O=all TerrariaServer.exe -configpath "$CONFIGPATH" -logpath "$LOGPATH" -config "$CONFIG" "$@"
```
Comment (ryansheehan) [inline tshock/bootstrap.sh:33]: If the `$CONFIG` variable is not set, will the server still launch?  Maybe we should default the value for the `$CONFIG` variable?

**Review by ryansheehan — COMMENTED** [review]
I think `$CONFIGPATH` and `$CONFIG` are two confusingly named variables.  I had to look them up on the TShock docs just to remind myself.

* `$CONFIGPATH` is where TShock looks to resolve the `config.json` file.  This file controls TShock settings
* `$CONFIG` (as it's set in this PR) is the configuration file that the `Terraria-Server.exe` uses to control aspects of the game (and some server options).

Ultimately I think TShock makes these differences confusing as there are overlaps between the two configurations.

I made several comments, and I would like your opinion.  

Overall, I'm not convinced we should introduce a volume for a single config file, and if it becomes a directory, all config should probably live there.  That would be a breaking change.  However, I think a dedicated config directory could be useful in some use cases.

The most common use case, though, is people use this container as an easy button to get their server up and running.  Containing all the config in a single location, with their save file, just makes things easier.  

Thank you for the contribution!  Let me know what you think.

File: tshock/serverconfig.txt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+#maxplayers=8
+
+#Set the port number
+#port=7777
+
+#Sets the world seed when using autocreate
+#seed=AwesomeSeed
+
+#Set the server password
+#password=p@55w0rd
```
[the remark this answers is not in the document]
Comment (velimir-97) [reply tshock/serverconfig.txt:27]: It appears that config.json takes president, I can remove the password field and leave a note redirecting the user to config.json to set a password, what do you think?

File: tshock/Dockerfile [hunk 5]
```diff
@@ -28,9 +29,10 @@ EXPOSE 7777 7878
 ENV CONFIGPATH=/root/.local/share/Terraria/Worlds
 ENV LOGPATH=/tshock/logs
 ENV WORLD_FILENAME=""
+ENV CONFIG=/config/serverconfig.txt
```
[the remark this answers is not in the document]
Comment (velimir-97) [reply tshock/Dockerfile:32]: I agree with this.

File: tshock/Dockerfile [hunk 6]
```diff
@@ -28,9 +29,10 @@ EXPOSE 7777 7878
...
 
 # Allow for external data
-VOLUME ["/root/.local/share/Terraria/Worlds", "/tshock/logs", "/plugins"]
+VOLUME ["/root/.local/share/Terraria/Worlds", "/tshock/logs", "/plugins", "/config"]
```
[the remark this answers is not in the document]
Comment (velimir-97) [reply tshock/Dockerfile:35]: We can keep everything in the `Worlds` folder. I have no issues with that.

File: tshock/bootstrap.sh (hunk 3 above)
Comment (velimir-97) [reply tshock/bootstrap.sh:33]: It is defaulted to `ENV CONFIG=/config/serverconfig.txt` line 32 tshock/Dockerfile currently, will update it to default to the `Worlds` dir of course.

**Pull request closed without merging**